router.go

```diff
@@ -65,7 +68,24 @@ func (h *router) Add(handler slog.Handler, predicates ...func(ctx context.Contex
 //
 //	A slog.Handler that implements the routing logic
 func (h *router) Handler() slog.Handler {
-	return Fanout(h.handlers...)
+	if h.firstMatch {
+		return FirstMatch(lo.Map(h.handlers, func(h slog.Handler, _ int) *RoutableHandler {
+			rh, ok := h.(*RoutableHandler)
+			if !ok {
+				panic(fmt.Sprintf("expected *RoutableHandler, got %T", h))
+			}
+			return rh
+		})...)
+	} else {
+		return Fanout(h.handlers...)
+	}
+}
+
+func (h *router) FirstMatch() *router {
+	return &router{
+		handlers:   h.handlers,
+		firstMatch: true,
```

Comment:
The `Add(...)` method must transmit the `firstMatch` value. It will allow unordered chaining of methods.

Reply:
done